src/libasr/pass/replace_symbolic.cpp

```diff
@@ -1762,6 +1764,28 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
             }
         }
     }
+
+    void visit_Return(const ASR::Return_t &x) {
+        SymbolTable* module_scope = current_scope->parent;
+        // freeing out variables
+        std::string new_name = "basic_free_stack";
+        ASR::symbol_t* basic_free_stack_sym = module_scope->get_symbol(new_name);
```

Comment:
Check if `module_scope->get_symbol(new_name)` is not `nullptr`.

Reply:
Do we return an error message if it's a `nullptr` ?

Reply:
Nope, Let's create and add the symbol:`basic_free_stack ` to the current scope.

Reply:
Actually I was thinking about it, as to in which cases would `basic_free_stack` be defined. And it goes like this
1) `basic_free_stack` is defined in `visit_Variable` and it only comes into play if the variable is of type `symbolic`
2) which means we only need basic_free_stack when we encounter a symbolic variable in our function.
3) The set keeping track of these symbolic variables is `symbolic_vars_to_free`. So we can say that at any points if `symbolic_vars_to_free` is non empty, it means we have encountered a symbolic variable while traversing through the pass, which means we have hit `visit_Variable` and hence `basic_free_stack` has been defined for such cases.